### PR TITLE
Remove OMZ model_tools dependency

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Remove instructions to install OMZ model_tools from Makefile

### Reason for changes

Vanishing unnecessary dependency

### Related tickets

136314

### Tests

N/A

### CODE REVIEW

**Comment by kshpv** [comment]
E2E ONNX 692
E2E TF 671
E2E Torch 735

File: Makefile [hunk 1]
```diff
@@ -64,8 +63,6 @@ test-examples-onnx:
 install-openvino-test:
 	pip install -U pip
 	pip install -e .
-	pip install "git+https://github.com/openvinotoolkit/open_model_zoo.git@37f60eb#egg=omz-tools&subdirectory=tools/model_tools"
-	pip install "git+https://github.com/openvinotoolkit/open_model_zoo.git@37f60eb#egg=accuracy_checker&subdirectory=tools/accuracy_checker"
```
Comment (KodiaqQ) [inline Makefile:68]: Why was the **accuracy_checker** removed?

File: tests/openvino/data/ac_configs/googlenet-v3-pytorch.yml
Comment (KodiaqQ) [inline tests/openvino/data/ac_configs/googlenet-v3-pytorch.yml:1]: Why these configurations were removed?

File: tests/shared/data/dataset_definitions.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    dataset_meta: voc12_without_background.json
+    postprocessing:
+      - type: resize_prediction_boxes
+    metrics:
+      - type: map
+        integral: 11point
+        ignore_difficult: True
+        presenter: print_scalar
+
+  - name: mapillary_vistas
```
Comment (KodiaqQ) [inline tests/shared/data/dataset_definitions.yml:92]: Are all the content is needed? For which models do we need the **mapillary_vistas** dataset?

File: Makefile (hunk 1 above)
Comment (kshpv) [reply Makefile:68]: It was accidentially added back in the https://github.com/openvinotoolkit/nncf/pull/2656/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R68 after removing it in https://github.com/openvinotoolkit/nncf/pull/2674

File: tests/openvino/data/ac_configs/googlenet-v3-pytorch.yml
Comment (kshpv) [reply tests/openvino/data/ac_configs/googlenet-v3-pytorch.yml:1]: They were fogotten to remove in https://github.com/openvinotoolkit/nncf/pull/2674. Before: they used at _test_sanity.py_, which was refactored to not use Accuracy Checker at all.

File: tests/shared/data/dataset_definitions.yml (hunk 2 above)
Comment (kshpv) [reply tests/shared/data/dataset_definitions.yml:92]: **mapillary_vistas** is used for _unet_mapillary_ model at `test_qat_segmentation.py`

File: Makefile (hunk 1 above)
Comment (KodiaqQ) [reply Makefile:68]: I still don't get why it was removed for the OV backend, since it contains regression accuracy checks for the OMZ models.

**Review by KodiaqQ — APPROVED** [review]

**Pull request merged**